{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import adafdr.method as md\n",
    "import adafdr.data_loader as dl\n",
    "import matplotlib.pyplot as plt\n",
    "import pickle\n",
    "%matplotlib inline\n",
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_folder = '/home/martin/NeuralFDR2/AdaFDRpaper/experiments_v1/temp_result'\n",
    "output_folder_fig = '/home/martin/NeuralFDR2/figures/fig_alg'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = {}\n",
    "# small gtex Adipose_Subcutaneous\n",
    "data_name = 'Adipose_Subcutaneous'\n",
    "data['small_gtex_Adipose_Subcutaneous'] = dl.data_small_gtex_chr21(opt=data_name)\n",
    "# small gtex Adipose_Visceral_Omentum\n",
    "data_name = 'Adipose_Visceral_Omentum'\n",
    "data['small_gtex_Adipose_Visceral_Omentum'] = dl.data_small_gtex_chr21(opt=data_name)\n",
    "# RNAseq: bottomly\n",
    "data['bottomly'] = dl.data_bottomly()\n",
    "# RNAseq: pasilla\n",
    "data['pasilla'] = dl.data_pasilla()\n",
    "# RNAseq: airway\n",
    "data['airway'] = dl.data_airway()\n",
    "# microbiome_enigma_ph\n",
    "data_path = '/data3/martin/AdaFDRpaper_data'\n",
    "file_path = data_path + '/microbiome/microbiome_enigma_ph'\n",
    "df_data = pd.read_csv(file_path, sep=',')\n",
    "p = df_data['p_val'].as_matrix()\n",
    "x = df_data[['ubiquity', 'mean_abun']].as_matrix()\n",
    "data['microbiome_enigma_ph'] = (p,x)\n",
    "# microbiome_enigma_al\n",
    "data_path = '/data3/martin/AdaFDRpaper_data'\n",
    "file_path = data_path + '/microbiome/microbiome_enigma_al'\n",
    "df_data = pd.read_csv(file_path, sep=',')\n",
    "p = df_data['p_val'].as_matrix()\n",
    "x = df_data[['ubiquity', 'mean_abun']].as_matrix()\n",
    "data['microbiome_enigma_al'] = (p,x)\n",
    "# proteomics\n",
    "data_path = '/data3/martin/AdaFDRpaper_data'\n",
    "file_path = data_path + '/proteomics/proteomics'\n",
    "df_data = pd.read_csv(file_path, sep=',')\n",
    "p = df_data['p_val'].as_matrix()\n",
    "x = df_data['x'].as_matrix()\n",
    "data['proteomics'] = (p,x)\n",
    "# fmri_auditory\n",
    "data_path = '/data3/martin/AdaFDRpaper_data'\n",
    "file_path = data_path + '/fmri/fmri_auditory'\n",
    "df_fmri = pd.read_csv(file_path, sep=',')\n",
    "p = df_fmri['p_val'].as_matrix()\n",
    "x = df_fmri['B_label'].as_matrix()\n",
    "data['fmri_auditory'] = (p,x)\n",
    "# fmri_imagination\n",
    "data_path = '/data3/martin/AdaFDRpaper_data'\n",
    "file_path = data_path + '/fmri/fmri_imagination'\n",
    "df_fmri = pd.read_csv(file_path, sep=',')\n",
    "p = df_fmri['p_val'].as_matrix()\n",
    "x = df_fmri['B_label'].as_matrix()\n",
    "data['fmri_imagination'] = (p,x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_list = list(data.keys())\n",
    "alpha_dic = {'small_gtex_Adipose_Subcutaneous': 0.01, \n",
    "             'small_gtex_Adipose_Visceral_Omentum': 0.01,\n",
    "             'bottomly':0.1,\n",
    "             'pasilla':0.1,\n",
    "             'airway':0.1,\n",
    "             'microbiome_enigma_ph': 0.2,\n",
    "             'microbiome_enigma_al': 0.2,\n",
    "             'proteomics': 0.1,\n",
    "             'fmri_auditory': 0.1,\n",
    "             'fmri_imagination': 0.1}\n",
    "fast_list = ['microbiome_enigma_ph','microbiome_enigma_al','proteomics',\n",
    "             'fmri_auditory','fmri_imagination']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "small_gtex_Adipose_Subcutaneous: rep 0\n",
      "## total rejection: 1469\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 1\n",
      "## total rejection: 1562\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 2\n",
      "## total rejection: 1446\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 3\n",
      "## total rejection: 1469\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 4\n",
      "## total rejection: 1426\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 5\n",
      "## total rejection: 1515\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 6\n",
      "## total rejection: 1570\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 7\n",
      "## total rejection: 1444\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 8\n",
      "## total rejection: 1469\n",
      "\n",
      "small_gtex_Adipose_Subcutaneous: rep 9\n",
      "## total rejection: 1469\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 0\n",
      "## total rejection: 1360\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 1\n",
      "## total rejection: 1367\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 2\n",
      "## total rejection: 1427\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 3\n",
      "## total rejection: 1400\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 4\n",
      "## total rejection: 1358\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 5\n",
      "## total rejection: 1490\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 6\n",
      "## total rejection: 1538\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 7\n",
      "## total rejection: 1393\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 8\n",
      "## total rejection: 1438\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum: rep 9\n",
      "## total rejection: 1484\n",
      "\n",
      "bottomly: rep 0\n",
      "## total rejection: 2144\n",
      "\n",
      "bottomly: rep 1\n",
      "## total rejection: 2145\n",
      "\n",
      "bottomly: rep 2\n",
      "## total rejection: 2169\n",
      "\n",
      "bottomly: rep 3\n",
      "## total rejection: 2139\n",
      "\n",
      "bottomly: rep 4\n",
      "## total rejection: 2187\n",
      "\n",
      "bottomly: rep 5\n",
      "## total rejection: 2178\n",
      "\n",
      "bottomly: rep 6\n",
      "## total rejection: 2138\n",
      "\n",
      "bottomly: rep 7\n",
      "## total rejection: 2148\n",
      "\n",
      "bottomly: rep 8\n",
      "## total rejection: 2181\n",
      "\n",
      "bottomly: rep 9\n",
      "## total rejection: 2152\n",
      "\n",
      "pasilla: rep 0\n",
      "## total rejection: 856\n",
      "\n",
      "pasilla: rep 1\n",
      "## total rejection: 826\n",
      "\n",
      "pasilla: rep 2\n",
      "## total rejection: 826\n",
      "\n",
      "pasilla: rep 3\n",
      "## total rejection: 804\n",
      "\n",
      "pasilla: rep 4\n",
      "## total rejection: 825\n",
      "\n",
      "pasilla: rep 5\n",
      "## total rejection: 837\n",
      "\n",
      "pasilla: rep 6\n",
      "## total rejection: 837\n",
      "\n",
      "pasilla: rep 7\n",
      "## total rejection: 807\n",
      "\n",
      "pasilla: rep 8\n",
      "## total rejection: 845\n",
      "\n",
      "pasilla: rep 9\n",
      "## total rejection: 810\n",
      "\n",
      "airway: rep 0\n",
      "## total rejection: 6050\n",
      "\n",
      "airway: rep 1\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-567c06bd620d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'%s: rep %d'\u001b[0m\u001b[0;34m%\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_name\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mi_rep\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m         res = md.adafdr_test(p, x, fast_mode=fast_mode, alpha=alpha, \n\u001b[0;32m---> 16\u001b[0;31m                              random_state=i_rep)\n\u001b[0m\u001b[1;32m     17\u001b[0m         \u001b[0mdecision\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi_rep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mres\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'decision'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m     \u001b[0mdecision_dic\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdata_name\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdecision\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/NeuralFDR2/adafdr/adafdr/method.py\u001b[0m in \u001b[0;36madafdr_test\u001b[0;34m(p_input, x_input, K, alpha, n_full, n_itr, qt_norm, covariate_type, h, verbose, output_folder, random_state, single_core, fast_mode)\u001b[0m\n\u001b[1;32m    483\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0msingle_core\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    484\u001b[0m         \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 485\u001b[0;31m         \u001b[0mres\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod_single_fold_wrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mY_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    486\u001b[0m         \u001b[0mres\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod_single_fold_wrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mY_input\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    487\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/NeuralFDR2/adafdr/adafdr/method.py\u001b[0m in \u001b[0;36mmethod_single_fold_wrapper\u001b[0;34m(data)\u001b[0m\n\u001b[1;32m    359\u001b[0m                                    \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mverbose\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput_folder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0moutput_folder\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    360\u001b[0m                                    \u001b[0mfold_number\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mfold_number\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrandom_state\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mrandom_state\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;31m\\\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 361\u001b[0;31m                                    f_write=f_write, fast_mode=fast_mode)\n\u001b[0m\u001b[1;32m    362\u001b[0m     \u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mw\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mmu\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0msigma\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mgamma\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtheta\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    363\u001b[0m     \u001b[0;31m# Test on the second fold\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/NeuralFDR2/adafdr/adafdr/method.py\u001b[0m in \u001b[0;36mmethod_single_fold\u001b[0;34m(p_input, x_input, K, alpha, n_full, n_itr, h, verbose, output_folder, fold_number, random_state, f_write, fast_mode)\u001b[0m\n\u001b[1;32m    670\u001b[0m         \u001b[0mloss\u001b[0m  \u001b[0;34m=\u001b[0m \u001b[0mloss1\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mloss2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    671\u001b[0m         \u001b[0;31m# Back propogation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 672\u001b[0;31m         \u001b[0mloss\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbackward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    673\u001b[0m         \u001b[0moptimizer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    674\u001b[0m         \u001b[0;31m# Record important quantities.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/torch/autograd/variable.py\u001b[0m in \u001b[0;36mbackward\u001b[0;34m(self, gradient, retain_graph, create_graph, retain_variables)\u001b[0m\n\u001b[1;32m    165\u001b[0m                 \u001b[0mVariable\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    166\u001b[0m         \"\"\"\n\u001b[0;32m--> 167\u001b[0;31m         \u001b[0mtorch\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mautograd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbackward\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgradient\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mretain_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcreate_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mretain_variables\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    168\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    169\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mregister_hook\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhook\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/torch/autograd/__init__.py\u001b[0m in \u001b[0;36mbackward\u001b[0;34m(variables, grad_variables, retain_graph, create_graph, retain_variables)\u001b[0m\n\u001b[1;32m     97\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     98\u001b[0m     Variable._execution_engine.run_backward(\n\u001b[0;32m---> 99\u001b[0;31m         variables, grad_variables, retain_graph)\n\u001b[0m\u001b[1;32m    100\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    101\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "n_rep = 10\n",
    "decision_dic = {}\n",
    "for data_name in data_list:\n",
    "    p = data[data_name][0]\n",
    "    x = data[data_name][1]\n",
    "    alpha = alpha_dic[data_name]\n",
    "    n_sample = p.shape[0]\n",
    "    decision = np.zeros([n_rep,n_sample], dtype=bool)\n",
    "    if data_name in fast_list:\n",
    "        fast_mode = True\n",
    "    else:\n",
    "        fast_mode = False\n",
    "    for i_rep in range(n_rep):\n",
    "        print('%s: rep %d'%(data_name,i_rep))\n",
    "        res = md.adafdr_test(p, x, fast_mode=fast_mode, alpha=alpha, \n",
    "                             random_state=i_rep)\n",
    "        decision[i_rep, :] = res['decision']\n",
    "    decision_dic[data_name] = decision"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fil = open(output_folder+'/res_stability.pickle','wb') \n",
    "pickle.dump(decision_dic, fil)\n",
    "fil.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "fil = open(output_folder+'/res_stability.pickle','rb') \n",
    "decision_dic = pickle.load(fil)\n",
    "fil.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "name_dic = {'small_gtex_Adipose_Subcutaneous': 'small GTEx:\\nAdipose_Subcutaneous', \n",
    "            'small_gtex_Adipose_Visceral_Omentum': 'small GTEx:\\nAdipose_Visceral_Omentum',\n",
    "            'bottomly': 'RNA-Seq: Bottomly',\n",
    "            'pasilla': 'RNA-Seq: Pasilla',\n",
    "            'airway': 'RNA-Seq: airway',\n",
    "            'microbiome_enigma_ph': 'microbiome: enigma_ph',\n",
    "            'microbiome_enigma_al': 'microbiome: enigma_al',\n",
    "            'proteomics': 'proteomics',\n",
    "            'fmri_auditory': 'fMRI: auditory',\n",
    "            'fmri_imagination': 'fMRI: imagination'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "small_gtex_Adipose_Subcutaneous\n",
      "Rep1 reject=1491, std=41\n",
      "Overlap=1408, std=25, percentage=94.5, std percent=1.7\n",
      "\n",
      "small_gtex_Adipose_Visceral_Omentum\n",
      "Rep1 reject=1396, std=96\n",
      "Overlap=1250, std=87, percentage=89.5, std percent=6.2\n",
      "\n",
      "bottomly\n",
      "Rep1 reject=2147, std=38\n",
      "Overlap=2010, std=36, percentage=93.6, std percent=1.7\n",
      "\n",
      "pasilla\n",
      "Rep1 reject=830, std=15\n",
      "Overlap=784, std=14, percentage=94.4, std percent=1.7\n",
      "\n",
      "airway\n",
      "Rep1 reject=6041, std=33\n",
      "Overlap=5871, std=46, percentage=97.2, std percent=0.8\n",
      "\n",
      "microbiome_enigma_ph\n",
      "Rep1 reject=119, std=8\n",
      "Overlap=104, std=5, percentage=87.8, std percent=4.8\n",
      "\n",
      "microbiome_enigma_al\n",
      "Rep1 reject=480, std=46\n",
      "Overlap=396, std=42, percentage=82.4, std percent=8.8\n",
      "\n",
      "proteomics\n",
      "Rep1 reject=408, std=18\n",
      "Overlap=366, std=14, percentage=89.6, std percent=3.5\n",
      "\n",
      "fmri_auditory\n",
      "Rep1 reject=1066, std=10\n",
      "Overlap=1033, std=11, percentage=96.9, std percent=1.1\n",
      "\n",
      "fmri_imagination\n",
      "Rep1 reject=2233, std=12\n",
      "Overlap=2178, std=13, percentage=97.6, std percent=0.6\n",
      "\n"
     ]
    }
   ],
   "source": [
    "temp_ind = np.arange(50)*2\n",
    "for data_name in data_list:\n",
    "    rep_1 = decision_dic[data_name][temp_ind,:]\n",
    "    rep_2 = decision_dic[data_name][temp_ind+1,:]\n",
    "    v_rej = np.sum(rep_1, axis=1)\n",
    "    v_overlap = np.sum((rep_1==True)&(rep_2==True), axis=1)\n",
    "    \n",
    "    print(data_name)\n",
    "    print('Rep1 reject=%d, std=%d'%(np.mean(v_rej),np.std(v_rej)))\n",
    "    print('Overlap=%d, std=%d, percentage=%0.1f, std percent=%0.1f'\n",
    "          %(np.mean(v_overlap),np.std(v_overlap), 100*np.mean(v_overlap)/np.mean(v_rej),\n",
    "            100*np.std(v_overlap)/np.mean(v_rej)))\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
